src/internal/streamstore/streamstore.go

```diff
@@ -0,0 +1,194 @@
+// streamstore implements helpers to store large
+// data streams in a repository
+package streamstore
```

Comment:
Two nits/questions about this package:

1/ should it live in /kopia?  The two packages seem fairly coupled.

2/ imo, the name feels a little off.  Any `fooStore` sounds like you're running a separate type of store (external to kopia, which is doubly easy to interpret from the package separation), rather than wrapping an existing store.  Not entirely sure what another name would be.  `streamBlob`?  `streamWriter`?  Maybe just reverse it: `storestream`?

Reply:
1) I thought about putting it in Kopia but the layering doesn't seem right there. I'd rather it be a consumer of Kopia (which is what it is right now - similar to model store).

2) I'd agree the name isn't perfect but I do think we should think of this as a separate store. I earlier had it as `metastore` which might be a better name. One of my thoughts as I was going through implementation was that this might be a good abstraction to plug in other external stores for this metadata. For example - if we wanted to write details directly into an external service.